standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -3288,6 +3291,190 @@ public List<Void> run(List<String> input) throws Exception {
     return true;
   }
 
+  // a helper function which will firstly get the current COLUMN_STATS_ACCURATE parameter on table level
+  // secondly convert the JSON String into map, and update the information in it, and convert it back to JSON
+  // thirdly update the COLUMN_STATS_ACCURATE parameter with the new value on table level using directSql
+  public long updateColumnStatsAccurateForTable(Table table, List<String> droppedCols) throws MetaException {
+    String currentValue = table.getParameters().get("COLUMN_STATS_ACCURATE");
+    if (currentValue == null) return 0;
+
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+
+      // Deserialize the JSON into a map
+      Map<String, Object> statsMap = mapper.readValue(currentValue, new TypeReference<Map<String, Object>>() {});
+
+      // Get the COLUMN_STATS object if it exists
+      Object columnStatsObj = statsMap.get("COLUMN_STATS");
+
+      if (columnStatsObj instanceof Map) {
+        Map<String, String> columnStats = (Map<String, String>) columnStatsObj;
+
+        boolean removeAll = (droppedCols == null || droppedCols.isEmpty());
+
+        if (removeAll) {
+          // Remove entire column stats
+          statsMap.remove("COLUMN_STATS");
+        } else {
+          // Remove only the dropped columns
+          for (String col : droppedCols) {
+            columnStats.remove(col.toLowerCase());
+          }
+          if (columnStats.isEmpty()) {
+            statsMap.remove("COLUMN_STATS");
+          }
+        }
+      }
+
+      // Serialize the map into a new JSON string
+      String updatedValue = mapper.writeValueAsString(statsMap);
+
+      // Update the COLUMN_STATS_ACCURATE parameter
+      return updateTableParam(table, "COLUMN_STATS_ACCURATE", currentValue, updatedValue);
+    } catch (Exception e) {
+      throw new MetaException("Failed to parse/update COLUMN_STATS_ACCURATE: " + e.getMessage());
+    }
+  }
+
+
+
+  public boolean updateColumnStatsAccurateForPartitions(String catName, String dbName, Table table,
+                                                     List<String> partNames, List<String> colNames) throws MetaException {
+    if (partNames == null || partNames.isEmpty()) {
+      return true;
+    }
+
+    ObjectMapper mapper = new ObjectMapper();
+
+    // If colNames is empty, then all the column stats of all columns should be deleted fetch all table column names
+    List<String> effectiveColNames;
+    if (colNames == null || colNames.isEmpty()) {
+      if (table.getSd().getCols() == null) {
+        effectiveColNames = new ArrayList<>();
+      } else {
+        effectiveColNames = table.getSd().getCols().stream()
+                .map(f -> f.getName().toLowerCase())
+                .collect(Collectors.toList());
+      }
+    } else {
+      effectiveColNames = colNames.stream().map(String::toLowerCase).collect(Collectors.toList());
+    }
+    List<String> finalColNames = effectiveColNames;
```

Comment:
We don't need `finalColNames` here as we are not updating `effectiveColNames` after this point.